### PR TITLE
Pyflyte run subcommand

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
`pyflyte run` a new command to help in the execution of single scripts

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
We define a new sub-command under `pyflyte` called `register` that takes as a parameter a path to a file and fast-registers a specific workflow defined in that file. 

The emphasis is on a single workflow in a single file, in other words, we will only copy the file specified by the user during invocation. For example, let's say we have the following directory structure:
```
    src
    ├── flyte
    │   ├── __init__.py
    │   └── workflows
    │       ├── example.py
    │       ├── another_example.py
    │       ├── yet_another_example.py
    │       └── __init__.py
```

If you `cd` into `src/flyte/workflows` and run `pyflyte run example.py` we will only copy `example.py` to the fast-registered code, which means that relative imports are not supported. Just to re-emphasize, the idea is to support getting up-and-running for a single file as quickly as possible.

The version used during fast-registration is produced by hashing the contents of the script.

After fast-registering the workflow, the sub-command schedules an execution of the workflow and waits for it to finish.

Let's say we have a file called `example.py` defined as:
```
import typing

from flytekit import task, workflow

@task
def say_hello(msg: str) -> str:
    return f"hello {msg}"

@workflow
def my_wf(message: str) -> str:
    res = say_hello(msg=message)
    return res
```

Here's an example of a run of the sub-command:
```
❯ pyflyte run flyte/workflows/example.py:my_wf
...
Go to http://localhost:30081/console/projects/flytesnacks/domains/development/executions/fe666e500cf0c4d18aef to see execution in the console.
```

### TODO
- [x] data proxy single url does not work on the sandbox. Being fixed as we speak.
- [x] data proxy generates the same url on consecutive calls if suffix is specified. Also being worked on as we speak.
- ~[ ] Handle input types other than strings.~
- ~[ ] Help message that explains how to use the sub-command, including how to specify inputs.~
- [ ] tests.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2241

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/remote/remote.py [hunk 1]
```diff
@@ -463,6 +464,29 @@ def register_workflow(
 
         return self.fetch_workflow(ident.project, ident.domain, ident.name, ident.version)
 
+    def register_workflow_script_mode(
```
Comment (kumare3) [inline flytekit/remote/remote.py:467]:
```suggestion
    def register_script(Workflow|Task)
```

File: flytekit/clis/sdk_in_container/run.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    project,
+    domain,
+    destination_dir,
+    image_config,
+    service_account,
+    wait_execution,
+    dump_snippet,
+):
+    """
+    Register command, a.k.a. script mode. It allows for a a single script to be registered and run from the command line
```
Comment (cosmicBboy) [inline flytekit/clis/sdk_in_container/run.py:109]: should this be `Run command, ...`

File: flytekit/clis/sdk_in_container/run.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+        if dump_snippet:
+            _dump_flyte_remote_snippet(execution, project, domain)
+    else:
+        inputs = _parse_workflow_inputs(
+            click_ctx,
+            wf_entity,
+            is_remote=False,
+        )
+        print(wf_entity(**inputs))
```
Comment (kumare3) [inline flytekit/clis/sdk_in_container/run.py:158]: click.secho(color=green)

File: flytekit/clis/sdk_in_container/run.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+def _parse_workflow_inputs(click_ctx, wf_entity, create_upload_location_fn: Optional[Callable] = None, is_remote=False):
+    args = {}
+    for i in range(0, len(click_ctx.args), 2):
+        argument = click_ctx.args[i][2:]
+        value = click_ctx.args[i + 1]
+
+        if argument not in wf_entity.interface.inputs:
+            raise FlyteValidationException(f"argument '{argument}' is not listed as a parameter of the workflow")
+
+        python_type = TypeEngine.guess_python_type(wf_entity.interface.inputs[argument].type)
```
Comment (kumare3) [inline flytekit/clis/sdk_in_container/run.py:184]: you can simply use - https://github.com/flyteorg/flytekit/blob/master/flytekit/core/workflow.py#L202 for getting the types and the names

File: flytekit/clis/sdk_in_container/run.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+)
+@click.option(
+    "-i",
+    "--image",
+    "image_config",
+    required=False,
+    multiple=True,
+    type=click.UNPROCESSED,
+    callback=ImageConfig.validate_image,
+    default=["ghcr.io/flyteorg/flytekit:py3.9-latest"],
```
Comment (kumare3) [inline flytekit/clis/sdk_in_container/run.py:69]: you can simply use DefaultImage

File: flytekit/remote/remote.py [hunk 6]
```diff
@@ -461,6 +474,61 @@ def register_workflow(
...
+        image_config: typing.Optional[ImageConfig] = None,
+        version: typing.Optional[str] = None,
+        project: typing.Optional[str] = None,
+        domain: typing.Optional[str] = None,
+        destination_dir: str = ".",
+        default_launch_plan: typing.Optional[bool] = True,
+        options: typing.Optional[Options] = None,
+    ) -> FlyteWorkflow:
+        """
+        Use this method to register a workflow via script mode.
```
Comment (kumare3) [inline flytekit/remote/remote.py:489]: not this PR, but we should add description of what `script mode` really means

**Review by kumare3 — APPROVED** [review]
